middleware/data_source_queries.py

```diff
@@ -163,8 +163,39 @@ def data_sources_results(conn):
     return results
 
 
-def data_sources_query(conn={}, test_query_results=[]):
-    results = data_sources_results(conn, "", "") if conn else test_query_results
+def needs_identification_results(conn):
+    cursor = conn.cursor()
+    data_source_approved_columns = [
+        f"data_sources.{approved_column}"
```

Comment:
As a minor nit, I would remove the data_sources prefix here -- since there's only one table involved in the SQL query, I think it's fairly clear what table each column is deriving from.